stock-tracker/src/features/search/redux/searchMiddleware.js

```diff
@@ -0,0 +1,40 @@
+import { setChartLoadingAction } from "../../chart/redux/actions";
+import { setLoadingOverviewAction } from "../../overview/redux/actions";
+import { setLoadingNewsAction } from "../../latestNews/redux/actions";
+import { setLoadingKeyStatsAction } from "../../keyStats/redux/actions";
+import { setLoadingPeersAction } from "../../topPeers/redux/actions";
+import { setSuggestionsAction } from "./actions";
+import { ADD_SYMBOL, ADD_SEARCH_INPUT } from "./actionTypes";
+import {
+  SYMBOL_INPUT,
+  SEARCH_INPUT,
+  SUGGESTIONS
+} from "../../../socket/eventTypes";
+
+export const searchMiddleware = ({
+  socketService
+}) => store => next => action => {
+  const result = next(action);
+  if (action.type === ADD_SYMBOL) {
+    socketService
+      .create()
+      .emit(
+        SYMBOL_INPUT,
+        store.getState().search.symbol,
+        store.getState().chart.chartTime
+      );
+    store.dispatch(setChartLoadingAction());
+    store.dispatch(setLoadingKeyStatsAction());
+    store.dispatch(setLoadingNewsAction());
+    store.dispatch(setLoadingOverviewAction());
+    store.dispatch(setLoadingPeersAction());
+  } else if (action.type === ADD_SEARCH_INPUT) {
+    socketService
+      .create()
+      .emit(SEARCH_INPUT, store.getState().search.searchInput);
+    socketService.create().on(SUGGESTIONS, suggestions => {
```

Comment:
**Memory leak**: Why're you creating a new `on(SUGGESTIONS, ...` handler here? This should be attached once to the socket, whereas it's currently being attached on every `ADD_SEARCH_INPUT` action

Reply:
And why is it not using your `socketService.createSocketSubscription` function?

Reply:
I thought it just receives the already created socket, have now attached it to a socket variable and reusing that variable.

Reply:
This hasn't fixed the memory leak. I wasn't referring to the use of the `socket` variable - That isn't a memory leak.
I'm referring to the fact that you're creating a new `.on` handler on every `ADD_SEARCH_INPUT` action, and never calling `.off` to remove it.

Reply:
My bad. I don't need this within the search input. Realized the subscription has already been created within subscriptions.js on start up.